Content.Server/Atmos/Monitor/Components/AirAlarmComponent.cs

```diff
@@ -0,0 +1,222 @@
+using static Content.Shared.Wires.SharedWiresComponent.WiresAction;
+
+namespace Content.Server.Atmos.Monitor.Components
+{
+    [RegisterComponent]
+    public class AirAlarmComponent : Component, IWires
+    {
+        [ComponentDependency] public readonly ApcPowerReceiverComponent? DeviceRecvComponent = default!;
+        [ComponentDependency] public readonly AtmosMonitorComponent? AtmosMonitorComponent = default!;
+        [ComponentDependency] public readonly DeviceNetworkComponent? DeviceNetComponent = default!;
+        [ComponentDependency] public readonly WiresComponent? WiresComponent = null;
+
+        private AirAlarmSystem? _airAlarmSystem;
+
+        [ViewVariables] public AirAlarmMode CurrentMode { get; set; }
+
+        // Remember to null this afterwards.
+        [ViewVariables] public IAirAlarmModeUpdate? CurrentModeUpdater { get; set; }
+
+        public override string Name => "AirAlarm";
+
+        public Dictionary<string, IAtmosDeviceData> DeviceData = new();
+
+        public HashSet<NetUserId> ActivePlayers = new();
+
+        public bool FullAccess = false;
+        public bool CanSync = true;
+
+        // <-- Wires -->
+
+        private CancellationTokenSource _powerPulsedCancel = new();
+        private int PowerPulsedTimeout = 30;
+
+        private enum Wires
+        {
+            // Cutting this kills power.
+            // Pulsing it disrupts power.
+            Power,
+            // Cutting this allows full access.
+            // Pulsing this does nothing.
+            Access,
+            // Cutting/Remending this resets ONLY from panic mode.
+            // Pulsing this sets panic mode.
+            Panic,
+            // Cutting this clears sync'd devices, and makes
+            // the alarm unable to resync.
+            // Pulsing this resyncs all devices (ofc current
+            // implementation just auto-does this anyways)
+            DeviceSync,
+            // This does nothing. (placeholder for AI wire,
+            // if that ever gets implemented)
+            Dummy
+        }
+
+        public void RegisterWires(WiresComponent.WiresBuilder builder)
+        {
+            builder.CreateWire(Wires.Power);
+            builder.CreateWire(Wires.Access);
+            builder.CreateWire(Wires.Panic);
+            builder.CreateWire(Wires.DeviceSync);
+            builder.CreateWire(Wires.Dummy);
+
+            UpdateWires();
+        }
+
+        public void UpdateWires()
+        {
+            if (_airAlarmSystem == null)
+                _airAlarmSystem = EntitySystem.Get<AirAlarmSystem>();
+
+            if (WiresComponent == null) return;
+
+            var powerLight = new StatusLightData(Color.Yellow, StatusLightState.On, "POWR");
+
+            if (PowerPulsed)
+                powerLight = new StatusLightData(Color.Yellow, StatusLightState.BlinkingFast, "POWR");
+            else if (PowerCut)
+                powerLight = new StatusLightData(Color.Yellow, StatusLightState.Off, "POWR");
```

Comment:
Not sure if this is the right syntax, but you get the picture.

```suggestion
            var pwrLightState = (PowerPulsed, PowerCut) switch {
                (true, false) => StatusLightState.BlinkingFast,
                (_, true) => StatusLightState.Off,
                (_, _) => StatusLightState.On
            };

            var powerLight = new StatusLightData(Color.Yellow, pwrLightState, "POWR");
```